File: python_programming_basics/for_loops/e_06_oscars.py
actor_name = input()
total_points = float(input())
judges_count = int(input())

for _ in range(judges_count):
    judge_name = input()
    judge_points = float(input())
    total_points += len(judge_name) * judge_points / 2

    if total_points > 1250.5:
        print(f"Congratulations, {actor_name} got a nominee for leading role with {total_points:.1f}!")
        break

else:
    needed_points = 1250.5 - total_points
    print(f"Sorry, {actor_name} you need {1250.5 - total_points:.1f} more!")

File: JavaScript Basics/04. For Loop/e-06-Oscars.js
function calculateActorPoints(input) {
    const actorName = input[0];
    let totalPoints = Number(input[1]);
    const numberOfJudges = Number(input[2]);

    for (let i = 0; i < numberOfJudges; i++) {
        const judgeName = input[3 + i * 2];
        const judgePoints = Number(input[4 + i * 2]);
        const pointsFromJudge = (judgeName.length * judgePoints) / 2;
        
        totalPoints += pointsFromJudge;

        if (totalPoints > 1250.5) {
            console.log(
                `Congratulations, ${actorName} got a nominee for leading role with ${totalPoints.toFixed(1)}!`
            );
            return;
        }
    }

    const neededPoints = (1250.5 - totalPoints).toFixed(1);
    console.log(
        `Sorry, ${actorName} you need ${neededPoints} more!`
    );
}
